geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/DataFrameFunctions.scala

```diff
@@ -29,90 +32,112 @@ object DataFrameFunctions extends SpatialEncoders {
+    /** Create a multi-line literal, encoded as a MultiPointUDT. */
+    def mLineLit(g: MultiLineString): TypedColumn[Any, MultiLineString] = udtlit(g, MultiLineStringUDT)
+    /** Create a multi-polygon literal, encoded as a MultiPolygonUDT. */
+    def mPolygonLit(g: MultiPolygon): TypedColumn[Any, MultiPolygon] = udtlit(g, MultiPolygonUDT)
+    /** create a geometry collection literal, encoded as a GeometryCollectionUDT. */
+    def geomCollLit(g: GeometryCollection): TypedColumn[Any, GeometryCollection] = udtlit(g, GeometryCollectionUDT)
+
     def st_geomFromWKT(wkt: Column): TypedColumn[Any, Geometry] =
       udfToColumn(ST_GeomFromWKT, constructorNames, wkt)
     def st_geomFromWKT(wkt: String): TypedColumn[Any, Geometry] =
-      udfToColumnLiterals(ST_GeomFromWKT, constructorNames, wkt)
+      st_geomFromWKT(lit(wkt))
 
     def st_geomFromWKB(wkb: Column): TypedColumn[Any, Geometry] =
       udfToColumn(ST_GeomFromWKB, constructorNames, wkb)
     def st_geomFromWKB(wkb: Array[Byte]): TypedColumn[Any, Geometry] =
-      udfToColumnLiterals(ST_GeomFromWKB, constructorNames, wkb)
+      st_geomFromWKB(lit(wkb))
 
     def st_lineFromText(wkt: Column): TypedColumn[Any, LineString] =
       udfToColumn(ST_LineFromText, constructorNames, wkt)
     def st_lineFromText(wkt: String): TypedColumn[Any, LineString] =
-      udfToColumnLiterals(ST_LineFromText, constructorNames, wkt)
+      st_lineFromText(lit(wkt))
 
     def st_makeBox2D(lowerLeft: Column, upperRight: Column): TypedColumn[Any, Geometry] =
       udfToColumn(ST_MakeBox2D, constructorNames, lowerLeft, upperRight)
     def st_makeBox2D(lowerLeft: Point, upperRight: Point): TypedColumn[Any, Geometry] =
-      udfToColumnLiterals(ST_MakeBox2D, constructorNames, lowerLeft, upperRight)
+      st_makeBox2D(pointLit(lowerLeft), pointLit(upperRight))
 
     def st_makeBBOX(lowerX: Column, upperX: Column, lowerY: Column, upperY: Column): TypedColumn[Any, Geometry] =
       udfToColumn(ST_MakeBBOX, constructorNames, lowerX, upperX, lowerY, upperY)
     def st_makeBBOX(lowerX: Double, upperX: Double, lowerY: Double, upperY: Double): TypedColumn[Any, Geometry] =
-      udfToColumnLiterals(ST_MakeBBOX, constructorNames, lowerX, upperX, lowerY, upperY)
+      st_makeBBOX(lit(lowerX), lit(upperX), lit(lowerY), lit(upperY))
 
     def st_makePolygon(lineString: Column): TypedColumn[Any, Polygon] =
-      udfToColumn(ST_MakePolygon, constructorNames, lineString)
+      st_polygon(lineString)
     def st_makePolygon(lineString: LineString): TypedColumn[Any, Polygon] =
-      udfToColumnLiterals(ST_MakePolygon, constructorNames, lineString)
+      st_polygon(lineString)
 
     def st_makePoint(x: Column, y: Column): TypedColumn[Any, Point] =
-      udfToColumn(ST_MakePoint, constructorNames, x, y)
+      st_point(x, y)
     def st_makePoint(x: Double, y: Double): TypedColumn[Any, Point] =
-      udfToColumnLiterals(ST_MakePoint, constructorNames, x, y)
+      st_point(x, y)
```

Comment:
Why don't you need `lit` here?

Reply:
`st_point` already has a `st_point(Double, Double)` form. But I could change it if it's confusing.